### ISSUE TITLE
SABnzbd-2.0.0Beta1-win64-bin.zip

### ISSUE THREAD

**Issue by OldskoolOrion** [sabnzbd/sabnzbd#854]
x64-package still deploys only 32-bit versions of unzip.exe and 7za.exe, while both have faster, more stable 64 bit versions of release you include of both.

Unrar x86 can be dropped as well - keeping them in isn't life threatening, but if it isn't there it can't be a point of failure either.

Most people prolly swap em out and delete the 32bit versions manually like me probably, every new release, so it's low priority since you guys have other stuff to do as well in live :-)  thanks in advance !

**Comment by Safihre** [comment]
They are there because on some older or more exotic Windows versions (Server versions) the 64bit par2/unrar sometimes gave problems. That's why we have the allow_64bit_tools Special setting to force 32bit versions. 
So no need for you to delete them, because Sab won't actually use them unless you enable that switch. 

But maybe indeed we don't need those anymore these days.

**Comment by OldskoolOrion** [comment]
ah.. ok :-)  but I do force them to go to the 64 bits since I delete the 32bit versions.. I've had my share of annoyance with the 32bit version of especially unrar 5.4.. the free one can really slow down.. no matter how high u set the privilige (io & cpu).  it held up a queue of already downloaded stuff unpacking something small.. and then it didnt a few hours later.. and then it did it again.. few days later I got tired of that and went for the 64b version  - that one hasn't let me down at all.. never displayed that behaviour.

I did see that info-zip's ftp is ... well.. it's up but logging on insta-errors when reading a dir out.. and empty too.. and they dont have the 64b version on their sucky sourceforge (thats soooo 2009 :P but yeh.. thats when the release was as well).  

I still had the sources so I compiled the whole bunch in 64bit for u - in case you might need them.. sha256 the 7z and had it tested by virustotal so you know i have no ill intent :-)
 
![virtotal 2017 03 12](https://cloud.githubusercontent.com/assets/13029885/23830171/96060e0c-0704-11e7-9846-23f5a13f659d.png)

I'll put the small archive + the sha256 hash on a webspace for now so you decide yourself if you trust it / want it (basically the info-zip - rest is easy to come by).  Don't think it's my place to just attach executable code in here.

[win.7z](http://beeldarchief.heemkundesjin.nl/cache_html/.quarantine/win.7z)
[win.sha256](http://beeldarchief.heemkundesjin.nl/cache_html/.quarantine/win.sha256)

Besides all this..  I respect your decision of course, since it has its reason(s). And I have a workaround anyway :-) Thanks for the swift answer !

**Comment by OldskoolOrion** [comment]
btw.. the .quarantine in the url has nothing to say.. just me having fun with scriptkiddies trying to exploit stuff and scan.. they love screwing it up for others but the moment they get near to a directory with a name like that, they get scared :-)

**Comment by Safihre** [comment]
> ah.. ok :-) but I do force them to go to the 64 bits since I delete the 32bit versions.. 

No you're not, we detect on startup that you run a 64bit OS and then we just ignore the 32bit executables whether you deleted them or not 😉 

I really appreciate you taking the time to compile these, but I don't think we will drop the 32bit stuff, since those executables only add about 1 MB to the total download-size and we didn't have any complaints so far. I also don't have the time right now to correctly implement this in the code :/

**Comment by thezoggy** [comment]
sab includes windows binaries for convenience sake and while we try to offer 32bit and 64bit if available but we prefer stability over it just showing '64' vs '32' to make someone happy.

We default to 64bit tools if the user os is 64bit (even if sab is being ran under 32bit python).
We do offer option to tell sab to use the 32bit versions of tools by unchecking config>special>allow_64bit_tools

You are correct to notice that on for the windows tools the included unzip and 7za are 32bit.
This has been noted previously:
> https://github.com/sabnzbd/sabnzbd/issues/619#issuecomment-246248342

In sab 2.0b1 we included 7-Zip 16.04 (2016-10-04), which is the latest.
And since forever now, the included version of unzip which is v6.00 from 2009 is the latest, there is no 64bit version from the devs.


I did find that there is a unzip 6.10b from 2010:
https://sourceforge.net/projects/infozip/files/unreleased%20Betas/UnZip%20betas/
```
Quick list of major changes in UnZip 6.10b:
 - Implement Windows full Unicode support. This support should be automatic now on NT and later platforms.
 - Add -I and -O options for setting ISO and OEM character sets, respectively, used by UnZip when doing character set translations. This support is based on the unzip60-alt-iconv-utf8 patch suggested in an Info-ZIP forum thread and uses the iconv library which must be available. These options are enabled using the USE_ICONV_MAPPING compiler macro. Suggestions welcome on how to improve this limited character translation support.
 - Various cleanup of VMS and Unicode code.
 - Better handling of the PKWare verification bit.
 - Fix bug in DLL code that prevented compiling.
 - Minor updates to the VB6 example code showing use of the DLL.
 - Implement Acorn port changes based on patch provided by J.L. that updates the port to use current compilers, increases allowed file path lengths to 1024 characters, adds directory attributes support, and updates documentation.
```

But we would have to compile it ourselves.. and if we did that.. we could in theory just make a 32bit and 64bit version...  but I don't know if there is any point in spending time to do so because of the lack of use for unzip with regards to usenet.. I've personally never seen sab need to run unzip so far in the last 5+ years. 

Now about 7z, it falls in a similar boat for me.. never have seen it used myself.
But including the 64bit executable is easily doable because it already exists.. but is it worth the extra bloat to the repo.

I grabbed the latest x64 7zip download and extracted to compare our 32 bit exe vs that one.. it has the same arguments but I noticed that ours mention `7za` while the other mentions `7z`:

```7-Zip (a) [32] 16.04 : Copyright (c) 1999-2016 Igor Pavlov : 2016-10-04
Usage: 7za <command> [<switches>...] <archive_name> [<file_names>...]
```
vs
```7-Zip [64] 16.04 : Copyright (c) 1999-2016 Igor Pavlov : 2016-10-04
Usage: 7z <command> [<switches>...] <archive_name> [<file_names>...]
```

looking into if there is a difference..  apparently on linux there is a light and regular version of the stand-alone version vs a wrapper version:
```
     7z uses plugins to handle archives.
    7za is a stand-alone executable. 7za handles fewer archive formats than 7z, but does not need any others.
    7zr is a stand-alone executable. 7zr handles fewer archive formats than 7z, but does not need any others. 7zr is a "light-version" of 7za that only handles 7z archives.
```

so long story short, if someone really wants x64 7za on windows they can just replace their exe with the one from the 7zip site.. for unzip.. if they want 64bit they need to go compile it themselves.

**Comment by OldskoolOrion** [comment]
Acknowledged :-)  i posted to already compiled 6.0 x64 - but agree that zip is not used that much anymore. Mainly 0day worked with it back in the day, but they decided to first zip and then rar the zips nowadays.
Then again rar knows zip.. 7zip knows both rar, zip and 7z. You could switch them all out for just 7zip if you want a lean and mean package.
I personally don't since rarlabs unpacking is just a tiny bit faster for v5 rars than 7z.  For me it's about performance - your reasons to keep it like it is are of course legit.

I think this one can be closed, since there is no real issue and kept like this for a reason :-)

**Comment by Safihre** [comment]
We thought about that too, but 7zip doesn't have complete support for all ways in which Rar archives can be sliced and encoded, at least not a year ago when I tried. 

So indeed closing this now because for the 32bit installer we need the 32bit executables anyway (so doesn't reduce installer size). 

We're always looking for people wanting to contribute, in case you're interested :)

**Comment by OldskoolOrion** [comment]
yeh thats true.. few kb more don't matter.. I'm stuck in a mindset of faster/leaner and less is better, because of my work and how I automate for myself.. Good that there's still people with focus on the enduser and that you point me to that :-) I try but forget and think everybody is a savy heavy duty professional/passionte geek like myself haha :-)  If you are going to support rar format, you have to be able to support all of it indeed.. people would think it'd look funny not seeing the word rar in one of other form for a file :-)   and.. I guess you have to leave room for tards trying to compress a movie at max level -m5 with 10% recovery records indeed :-)

And yes I'm interested... but for the moment I think between work and projects I'm already in, my time would be spread too thin : if I promise something, I like (read compulsive have to) to follow thru and actually do alot :-)  When 2 other projects start winding down after the next BSD port-release cycle, I'll probably do have spare time to invest :-)   (but better not hope I do frontend/ux stuff - i'm pretty generalistic but design is not something I do well :P)
